code/lib/types/src/modules/story.ts

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import type { SynchronousPromise } from 'synchronous-promise';
+import type { Renderer, ProjectAnnotations as CsfProjectAnnotations } from '@storybook/csf';
+
+import type {
+  ComponentAnnotations,
+  ComponentId,
+  ComponentTitle,
+  LegacyStoryFn,
+  PartialStoryFn,
+  Path,
+  StoryAnnotations,
+  StoryContext,
+  StoryContextForEnhancers,
+  StoryContextForLoaders,
+  StoryFn,
+  StoryId,
+  StoryIdentifier,
+  StoryName,
+  StrictArgTypes,
+  StrictGlobalTypes,
+} from './csf';
+
+// Internal to preview, exported until preview package consolidation
+export type Store_PromiseLike<T> = Promise<T> | SynchronousPromise<T>;
+export type PreparedStorySpecifier = StoryId | { name: StoryName; title: ComponentTitle } | '*';
```

Comment:
I still need to figure out how to move these out of here (small issue with `SynchronousPromise` to be resolved).

Reply:
I think this could be moved to the preview-api package now?

Reply:
Yes, that is my plan.